### ISSUE TITLE
Node version limits

### ISSUE THREAD

**Issue by zenicek** [awinogrodzki/next-firebase-auth-edge#47]
Hey,

Thanks for creating this library, its great help to managing auth. Is there any particular reason for limiting the upper versions of node/npm/yarn? I have checked the main dependencies and they don't seem to have any limitation on their end?

We run on node 19 and npm 9.5 and obviously we would have downgrade our whole app in order to use the dependency, which doesn't seem like a sensible option.

I am happy to check out and submit the PR.

`  "engines": {
    "node": ">=16.0.0 <19.0.0",
    "npm": ">=8.0.0 <9.0.0",
    "yarn": ">=1.22.0 <2.0.0"
  },`

**Comment by awinogrodzki** [comment]
Hey @zenicek! Thanks for reporting. I'm really glad you find this library helpful. Good question. I guess it was just a precaution, as I haven't tested the library on other node versions aside from those mentioned in "engine".

I'd be happy to merge the PR :-)

**Comment by awinogrodzki** [comment]
Hey @zenicek,

I've been able to test the library on Node 19 and it's working fine.

The Node.js 19 support has been introduced in `next-firebase-auth-edge@0.6.2`

Cheers 🎉

**Comment by zenicek** [comment]
Awesome, thanks for updating. I was looking into it, but when updated on the higher version, my tests started to fail, and I didn't get much time to dig deeper into it, but it was related to the `jest-setup.ts` and reassigning crypto as a read-only property. Did you manage to solve it?

I had to change it to something like this, but then I got another set of errors this time in tests itself :D :

`Object.defineProperty(global, 'crypto', {
  writable: true,
  value: new Crypto(),
});`
